python/ray/tests/test_object_spilling.py

```diff
@@ -240,6 +240,107 @@ def task():
+)
+def test_object_spilling_enabled_cli(call_ray_start, shutdown_only):
+    ray_context = ray.init(address=call_ray_start)
+
+    # Make sure the default spill directory is empty after running the workload.
+    run_basic_workload()
+    assert is_dir_empty(
+        Path(ray._private.worker._global_node._session_dir), ray_context["node_id"]
+    )
+
+
+def test_object_spilling_enabled_ray_params_and_system_config(shutdown_only):
+    # Set the object spilling enabled through both ray init and system config.
+    # The ray init should take precedence.
+    ray_context = ray.init(
+        num_cpus=0,
+        object_store_memory=75 * 1024 * 1024,
+        object_spilling_enabled=False,
+        _system_config={"automatic_object_spilling_enabled": True},
+    )
+    assert not ray._private.worker._global_node._config[
+        "automatic_object_spilling_enabled"
+    ]
+
+    # Make sure config is not initalized if spilling is not enabled.
+    assert "object_spilling_config" not in ray._private.worker._global_node._config
+
+    # Make sure the default spill directory is empty after running the workload.
+    run_basic_workload()
+    assert is_dir_empty(
+        Path(ray._private.worker._global_node._session_dir), ray_context["node_id"]
+    )
+
+
+def test_object_spilling_enabled_system_config_and_env_var(shutdown_only):
+    # Set the object spilling enabled through both system config and env var.
+    # The system config should take precedence.
+    os.environ["RAY_automatic_object_spilling_enabled"] = "true"
+    ray_context = ray.init(
+        num_cpus=0,
+        object_store_memory=75 * 1024 * 1024,
+        _system_config={"automatic_object_spilling_enabled": False},
+    )
+    assert not ray._private.worker._global_node._config[
+        "automatic_object_spilling_enabled"
+    ]
+
+    # Make sure config is not initalized if spilling is not enabled.
+    assert "object_spilling_config" not in ray._private.worker._global_node._config
+
+    # Make sure the default spill directory is empty after running the workload.
+    run_basic_workload()
+    assert is_dir_empty(
+        Path(ray._private.worker._global_node._session_dir), ray_context["node_id"]
+    )
+
+
+def test_object_spilling_enabled_env_var(shutdown_only):
+    os.environ["RAY_automatic_object_spilling_enabled"] = "false"
+    ray_context = ray.init(num_cpus=0, object_store_memory=75 * 1024 * 1024)
+    assert not ray._private.worker._global_node._config[
+        "automatic_object_spilling_enabled"
+    ]
+
+    # Make sure config is not initalized if spilling is not enabled.
+    assert "object_spilling_config" not in ray._private.worker._global_node._config
+
+    # Make sure the default spill directory is empty after running the workload.
+    run_basic_workload()
+    assert is_dir_empty(
+        Path(ray._private.worker._global_node._session_dir), ray_context["node_id"]
+    )
+
+    # clear env var
+    os.environ.pop("RAY_automatic_object_spilling_enabled")
```

Comment:
I believe we use `monkeypatch.setenv` elsewhere so it's cleaned up after the test (even if an exception is raised)